mesa/visualization/components/matplotlib.py

```diff
@@ -238,32 +250,14 @@ def portray(space):
     space_ax.set_ylim(space.y_min - y_padding, space.y_max + y_padding)
 
     # Portray and scatter the agents in the space
-    _split_and_scatter(portray(space), space_ax)
+    _split_and_scatter(agent_data, space_ax)
 
 
 def _draw_voronoi(space, space_ax, agent_portrayal):
-    def portray(g):
-        x = []
-        y = []
-        s = []  # size
-        c = []  # color
-
-        for cell in g.all_cells:
-            for agent in cell.agents:
-                data = agent_portrayal(agent)
-                x.append(cell.coordinate[0])
-                y.append(cell.coordinate[1])
-                if "size" in data:
-                    s.append(data["size"])
-                if "color" in data:
-                    c.append(data["color"])
-        out = {"x": x, "y": y}
-        out["s"] = s
-        if len(c) > 0:
-            out["c"] = c
-
-        return out
+    """Draw agents in a Voronoi space."""
+    agent_data = _get_agent_data(space, agent_portrayal)
```

Comment:
This won't work. This method is for plotting the experimental Voronoi space. 

The `_get_agent_data` method presupposes an old style space because it does `for agents, pos in space.coord_iter():`.